google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/apis/CloudApiManagementConfirmationDialogTest.java

```diff
@@ -166,4 +172,111 @@ public void roleTable_whenRolesExist_isPopulated_andAllSelectedByDefault() {
               assertThat(dialog.getSelectedRoles()).containsExactlyElementsIn(roles);
             });
   }
+
+  @Test
+  public void serviceAccount_whenFieldsCorrectlyPopulated_showsNoValidationError() {
+    ApplicationManager.getApplication()
+        .invokeAndWait(
```

Comment:
What usually happens when a test running inside of `invokeAndWait` thread fails? Will it properly propagate failed assertions to main test thread?

Reply:
yes it will